tests/tests/test_registries.py

```diff
@@ -0,0 +1,181 @@
+        self.registry.update_related(instance)
+
+        doc_d1.get_instances_from_related.assert_called_once_with(instance)
+        doc_d1.update.assert_not_called()
+
+    def test_delete_instance(self):
+        doc_a3 = self._generate_doc_mock(self.ModelA, self.index_1, _ignore_signals=True)
+
+        instance = self.ModelA()
+        self.registry.delete(instance)
+
+        self.assertFalse(doc_a3.update.called)
+        self.assertFalse(self.doc_b1.update.called)
+        self.doc_a1.update.assert_called_once_with(instance, "delete")
+        self.doc_a2.update.assert_called_once_with(instance, "delete")
+
+    def test_delete_related_instance(self):
+        """
+        Test that update is called when related_instance is defined.
+
+        doc_d1 is a related doc for instance. If we call delete_related on
+        instance, then update should be called with related_instance.
+        """
+        doc_d1 = self._generate_doc_mock(self.ModelD, self.index_1, _related_models=[self.ModelE])
+
+        instance = self.ModelE()
+        related_instance = self.ModelD()
+
+        doc_d1.get_instances_from_related.return_value = related_instance
+        self.registry.delete_related(instance)
+
+        doc_d1.get_instances_from_related.assert_called_once_with(instance)
+        doc_d1.update.assert_called_once_with(related_instance)
+
+    def test_delete_related_instance_not_defined(self):
+        """
+        Test that update is not called if related not defined.
+
+        If we try to call delete_related on instance when
+        get_instances_from_related returns None on doc_d1, then update
+        should not be called on doc_d1.
+        """
+        doc_d1 = self._generate_doc_mock(_model=self.ModelD, index=self.index_1, _related_models=[self.ModelE])
+
+        instance = self.ModelE()
+
+        doc_d1.get_instances_from_related.side_effect = ObjectDoesNotExist(Mock(return_value=None))
+        self.registry.delete_related(instance)
+
+        doc_d1.get_instances_from_related.assert_called_once_with(instance)
+        doc_d1.update.assert_not_called()
+
+    @mock.patch("django_opensearch_dsl.apps.DODConfig.autosync_enabled")
+    def test_delete_related_autosync_disabled(self, autosync_mock):
+        """If autosync is disabled, delete_related should just return."""
+
+        autosync_mock.return_value = False
+
+        doc_d1 = self._generate_doc_mock(self.ModelD, self.index_1, _related_models=[self.ModelE])
+
+        instance = self.ModelE()
+
+        doc_d1.get_instances_from_related.return_value = None
+        self.registry.delete_related(instance)
+
+        doc_d1.get_instances_from_related.assert_not_called()
+        doc_d1.update.assert_not_called()
+
+    def test_autosync(self):
+        settings.OPENSEARCH_DSL_AUTOSYNC = False
+
+        instance = self.ModelA()
+        self.registry.update(instance)
+        self.assertFalse(self.doc_a1.update.called)
+
+        settings.OPENSEARCH_DSL_AUTOSYNC = True
```

Comment:
Please use Django's `django.test.override_settings` either as a decorator or context manager instead of modifying `settings` directly.